ESP8266Interface.cpp

```diff
@@ -182,6 +196,19 @@ int8_t ESP8266Interface::get_rssi()
 
 int ESP8266Interface::scan(WiFiAccessPoint *res, unsigned count)
 {
+    _esp.setTimeout(ESP8266_MISC_TIMEOUT);
+
+    if (_esp.get_firmware_version() != ESP8266_VERSION) {
+        debug("ESP8266: ERROR: Firmware incompatible with this driver.\
+               \r\nUpdate to v%d - https://developer.mbed.org/teams/ESP8266/wiki/Firmware-Update\r\n",ESP8266_VERSION);
+        return NSAPI_ERROR_DEVICE_ERROR;
+    }
+
+    _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
+    if(!_esp.isConnected() && !_esp.startup(1)) {
```

Comment:
Why the modes given as parameters for startup are differing between connect and scan?

Reply:
I don't quite even understand that why in connect softap+station is used when station only would be sufficient.